src/pages/[post].tsx

```diff
@@ -447,6 +536,12 @@ function PostPlayer({
   const {mutateAsync: addProgressToLesson} =
     trpc.progress.addProgressToLesson.useMutation()
 
+  const {data: viewer} = trpc.user.current.useQuery()
+
+  const isPro = post.fields.access === 'pro'
+  const canView =
+    !isPro || (isPro && Boolean(viewer) && Boolean(viewer?.is_pro))
```

Comment:
canView logic

`(isPro && Boolean(viewer) && Boolean(viewer?.is_pro)` -> if the lesson has `pro` access level, `viewer` needs to be defined and viewer `is_pro` attribute set to true